homeassistant/components/subaru/__init__.py

```diff
@@ -0,0 +1,213 @@
+        except SubaruException as err:
+            raise UpdateFailed(err.message) from err
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name=COORDINATOR_NAME,
+        update_method=async_update_data,
+        update_interval=timedelta(
+            seconds=entry.options.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
+        ),
+    )
+
+    await coordinator.async_refresh()
+
+    hass.data.get(DOMAIN)[entry.entry_id] = {
+        ENTRY_CONTROLLER: controller,
+        ENTRY_COORDINATOR: coordinator,
+        ENTRY_VEHICLES: vehicle_info,
+        ENTRY_LISTENER: entry.add_update_listener(update_listener),
+    }
+
+    for component in SUPPORTED_PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, component)
+        )
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Unload a config entry."""
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(entry, component)
+                for component in SUPPORTED_PLATFORMS
+            ]
+        )
+    )
+    if unload_ok:
+        hass.data[DOMAIN][entry.entry_id][ENTRY_LISTENER]()
+        hass.data[DOMAIN].pop(entry.entry_id)
+
+    return unload_ok
+
+
+async def update_listener(hass, config_entry):
+    """Update when config_entry options update."""
+    data = hass.data[DOMAIN][config_entry.entry_id]
+    controller = data[ENTRY_CONTROLLER]
+    coordinator = data[ENTRY_COORDINATOR]
+
+    old_update_interval = controller.get_update_interval()
+    old_fetch_interval = coordinator.update_interval
+
+    new_update_interval = config_entry.options.get(
+        CONF_HARD_POLL_INTERVAL, DEFAULT_HARD_POLL_INTERVAL
+    )
+    new_fetch_interval = config_entry.options.get(
+        CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL
+    )
+
+    if old_update_interval != new_update_interval:
+        _LOGGER.debug(
+            "Changing update_interval from %s to %s",
+            old_update_interval,
+            new_update_interval,
+        )
+        controller.set_update_interval(new_update_interval)
+
+    if old_fetch_interval != new_fetch_interval:
+        _LOGGER.debug(
+            "Changing fetch_interval from %s to %s",
+            old_fetch_interval,
+            new_fetch_interval,
+        )
+        coordinator.update_interval = timedelta(seconds=new_fetch_interval)
+
+
+async def subaru_update(vehicle_info, controller):
+    """
+    Update local data from Subaru API.
+
+    Subaru API calls assume a server side vehicle context
+    Data fetch/update must be done for each vehicle
+    """
+    data = {}
+
+    for vehicle in vehicle_info.values():
+        vin = vehicle[VEHICLE_VIN]
+
+        # Active subscription required
+        if not vehicle[VEHICLE_HAS_SAFETY_SERVICE]:
+            continue
+
```

Comment:
If their subscription lapses for a vehicle this will return data without the `vin` set which will likely result in some breakage.

Set the `available` property to `False` in the entity when this happens.

Reply:
`available` property added to `SubaruSensor`